examples/raspberry-pi-threads.rs

```diff
@@ -0,0 +1,57 @@
+use navigator_rs::{Navigator, UserLed};
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread::{self, sleep};
+use std::time::Duration;
+
+fn main() {
+    println!("Creating your SensorData cache!");
+    let sensor_data = Arc::new(RwLock::new(navigator_rs::SensorData::default()));
+
+    println!("Creating your navigator module!");
+    let nav = Arc::new(Mutex::new(Navigator::new()));
+
+    println!("Setting up your navigator, ahoy!");
+    nav.lock().unwrap().init();
+
+    // This code block creates a thread that updates the `sensor_data` periodically.
+    let sensor_data_cloned = sensor_data.clone();
+    let nav_cloned = nav.clone();
+    let _ = thread::spawn(move || {
```

Comment:
Oh, just to be cooler, you can name the thread:
```Rust
std::thread::Builder::new()
    .name("The coolest thread ever".into())
    .spawn(move || {
        todo!();
    })
    .expect("Failed to spawn the coolest thread ever");
```

Reply:
did it for both threads, nice!